pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -215,22 +343,45 @@ func (ic *ifConfigurator) checkContainerInterface(
 			containerID, containerNetns, err)
 		return nil, err
 	}
-	return contVeth, nil
+	return networkInterface, nil
 }
 
-func validateContainerInterface(intf *current.Interface) (*vethPair, error) {
-	link, err := validateInterface(intf, true)
+func validateContainerVFInterface(intf *current.Interface, sriovVFDeviceID string) (netlink.Link, error) {
+	link, err := validateInterface(intf, true, vf)
 	if err != nil {
 		return nil, err
 	}
+	netdevices, _ := sriovnet.GetNetDevicesFromPci(sriovVFDeviceID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find netdevice to PCI address %s error %v", sriovVFDeviceID, err)
+	}
+	// the check just make sure that the SR-IOV VF netdevice is not in the host namespace
+	// the GetNetDevicesFromPci is using linux sysfs to find the VF netdevice
+	// the method is running in container network namespace, but we are still in the host filesystem
```

Comment:
I'm confused by "we are still in the host filesystem" since we are running inside a container. I think maybe you mean something along the lines of "the host's sysfs is mounted to the container because it is privileged" (I do not remember if this is accurate but is this what you mean?). Probably worth clarifying.

Reply:
I wasn't referring to the target Pod here. I was referring to the antrea-agent container in which this code is being executed. It has its own filesystem but I believe because it is privileged sysfs is the same as the host.

Reply:
@moshe010 can you clarify?